apps/studio/src/pages/api/trpc/[trpc].ts

```diff
@@ -24,7 +24,7 @@ export default trpcNext.createNextApiHandler({
    * Enable query batching
    */
   batching: {
-    enabled: true,
+    enabled: false,
```

Comment:
the `maxUrlLength` shown in the docs is set on the client side leh, not server side. 

> worst case is we eat the length clock cycles (trpc sends request to backend, request is empty, terminate immediately). because this length will be small, we still enable trpc to do some batching but mitigate risk of unbounded url length and hence, ddos attacks. however, this still means that, for example, each http request can actually contain up to length trpc requests. but because this means for the user, each request is actually more efficient, i think this might be ok

its slightly deeper than that. trpc v10 does not allow you to set length on the server side, so this implementation has to be handrolled + prob need to pin trpc version in our package.json in case they change how they do batching, then this handrolled limit might not work as intended even for minor changes for v10. 

for egazette, the worst case observed was a dos, since this was not enforced at the server side. if yall feel, this is not an ok tradeoff + willing to handroll implementation (which is quite doable, but i dont know how to test with confidence), will close this pr then yall handle it closer to vapt

Reply:
ah yea you're right. i think another alternative that we could think about is setting an explicit limit using a rate limiter middleware but this is still `limit * url length` in the worst case. lemme give it abit mroe thought

Reply:
> ah yea you're right. i think another alternative that we could think about is setting an explicit limit using a rate limiter middleware but this is still limit * url length in the worst case. lemme give it abit mroe thought

Sadly, that too does not work. Since this is not a valid route, we cannot attach a rate limiter to this. The DOS occurs as TRPC is trying to return ROUTE_NOT_FOUND errors